### PR TITLE
Simple integration with Ramp Network

### PR DESCRIPTION
**Pull request by gkapkowski**
ezgif com-video-to-gif(1)

(ignore the widget position and size, I have tiling window manger)

### CODE REVIEW

**Comment by chaitanyapotti** [comment]
Hi, 
we provide dapps to invoke the payment API directly from the [torus-embed](https://github.com/torusresearch/torus-embed) using this [API](https://docs.tor.us/api-reference/topup)
The current partners for this API are moonpay, wyre and coindirect. 
If you wish to be integrated here, there are certain requirements we have.
- Not use a sdk
- Provide a quote API to get a quote
- Have a hosted website which will be shown in popup triggered by torus
- Allow form inputs in the hosted website to be picked up from query params of the url
- Have a redirect url support to redirect to after the payment flow is completed
- Ping an endpoint of torus to notify about the tx details + status(webhook)

File: app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+<script>
+import { post } from '../../../utils/httpHelpers'
+import config from '../../../config'
+import log from 'loglevel'
+import { RampInstantSDK } from '@ramp-network/ramp-instant-sdk'
+
+export default {
+  methods: {
+    showWidget() {
+      console.log(arguments)
```
Comment (iamonuwa) [inline app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue]: Remove this console

File: app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      new RampInstantSDK({
+        hostAppName: 'Tor.us',
+        hostLogoUrl: 'https://tor.us/assets/img/torus-logo.svg',
+        userAddress: this.$store.state.selectedAddress
+      })
+        //.on('*', event => console.log(event))
```
Comment (iamonuwa) [inline app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue]: Remove this if not in use

Comment (gkapkowski) [reply app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue]: Done

File: app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue (hunk 1 above)
Comment (gkapkowski) [reply app/src/containers/WalletTopup/WalletTopupRampNetwork/WalletTopupRampNetwork.vue]: Done

**Comment by gkapkowski** [comment]
@chaitanyapotti I've updated the integration with our API call. Can you tell me more about calling you webhook?

**Comment by iamonuwa** [comment]
@gkapkowski just asking though, your logo on our dark theme looks off abit. Do you think you can fix that?

**Comment by chaitanyapotti** [comment]
We will create an endpoint for you. (rampnetwork-api.tor.us/transaction)
Please make a POST request to that endpoint with the tx object.
Include a signature created from tx object as a header in the POST request which we can verify with a secret API Key which is known to both of us.


Furthermore, please merge current develop of torus into your branch

Thanks,

**Comment by gkapkowski** [comment]
@iamonuwa I've fixed the logo. Thanks for pointing this out!

@chaitanyapotti I've added webhook configuration to the widget. The description and examples of tx object can be found in [Webhooks Guide](https://docs-instant.ramp.network/guide/events/webhooks/) and [Webhooks Reference](https://docs-instant.ramp.network/reference/webhooks/), We will add signature header this week. How will we receive secret?

**Comment by chaitanyapotti** [comment]
@gkapkowski  The secret would be an API Key generated by you and shared with us.

**Comment by gkapkowski** [comment]
@chaitanyapotti we decided to implement signatures based on ECDSA (the same as ethereum signatures) so that we can share our public key in documentation and the verification code on your side can use it. Will that work for you?
It's much easier than sharing secrets and you already work with libraries that have this verification mechanism implemented.

**Comment by chaitanyapotti** [comment]
That would totally work.
Thanks

**Comment by chaitanyapotti** [comment]
@gkapkowski 
We are ready for the merge as the backend is ready. Can you rebase your repo with the current develop branch?

**Comment by gkapkowski** [comment]
@chaitanyapotti I did rebase, please verify.

**Pull request merged**